src/main/java/net/minecraftforge/common/crafting/NBTIngredient.java

```diff
@@ -78,17 +94,17 @@ public JsonElement toJson()
 
         @Override
         public NBTIngredient parse(FriendlyByteBuf buffer) {
-            return new NBTIngredient(buffer.readItem());
+            return new NBTIngredient(buffer.readItem(), buffer.readBoolean());
         }
 
         @Override
         public NBTIngredient parse(JsonObject json) {
-            return new NBTIngredient(CraftingHelper.getItemStack(json, true));
+            return new NBTIngredient(CraftingHelper.getItemStack(json, true), GsonHelper.getAsBoolean(json, "exact_match", false));
         }
 
         @Override
         public void write(FriendlyByteBuf buffer, NBTIngredient ingredient) {
-            buffer.writeItem(ingredient.stack);
+            buffer.writeItem(ingredient.stack).writeBoolean(ingredient.exactMatch);
```

Comment:
There is no real reason to chain these calls compared to adding buffer.writeBoolean on the next line which is a bit more readable.